### PR TITLE
Update security overview with additional recommendations

### PR DESCRIPTION
**Pull request by techdocsmith**
Update security overview with additional recommendations for improved security

### Description

Adds best practice to clarify Druid behavior and necessary measures to run Druid in a secure environment.

This PR has:
- [x] been self-reviewed.

cc: @2bethere , @suneet-s

### CODE REVIEW

File: docs/operations/security-overview.md [hunk 1]
```diff
@@ -23,66 +23,109 @@ title: "Security overview"
...
+### Generating keys
+Before you enable TLS in Druid, generate the keystore and truststore. When one Druid process, e.g. Broker, contacts another Druid process , e.g. Historical, the first service is a client for the second service, considered the server.
 
-The configuration settings mentioned below are primarily located in the `common.runtime.properties` file. Note that you need to make the configuration changes on each Druid server in the cluster. 
+The client uses a trustStore that contains certificates trusted by the client. For example, the Broker.
 
+The server uses a keyStore that contains private keys and certificate chain used to securely identify itself.
 
-## Enable TLS
+The following example demonstrates how to use Java keytool to generate the keyStore for the server and then create a trustStore to trust the key for the client:
```
Comment (suneet-s) [inline docs/operations/security-overview.md:67]: nit: Since keytool is a tool, perhaps it should be escape quotes. Maybe we should capitalize KeyStore since that's how I see it show up when I search google?

```suggestion
The following example demonstrates how to use Java `keytool` to generate the KeyStore for the server and then create a trustStore to trust the key for the client:
```

File: docs/operations/security-overview.md [hunk 2]
```diff
@@ -23,66 +23,109 @@ title: "Security overview"
...
 
-The first step in securing Druid is enabling TLS. You can enable TLS to secure external client connections to Druid as well as connections between cluster nodes. 
+1. Generate the keyStore with Java keytool:
```
Comment (suneet-s) [inline docs/operations/security-overview.md:69]: nit: Then the spelling file won't need keytool to be added to it
```suggestion
1. Generate the keyStore with Java `keytool`:
```

File: docs/operations/security-overview.md [hunk 3]
```diff
@@ -23,66 +23,109 @@ title: "Security overview"
...
-To implement Druid security, you configure authenticators and authorizers. Authenticators control the way user identities are verified, while authorizers map the authenticated users (via user roles) to the datasources they are permitted to access. Consequently, implementing Druid security also involves considering your datasource scheme, since that scheme represents the granularity at which data access permissions are allocated. 
 
-The following graphic depicts the course of request through the authentication process: 
+## Best practices
 
 
-![Druid security check flow](../assets/security-model-1.png "Druid security check flow") 
+* Run Druid as an unprivileged Unix user. Do not run Druid as the root user.
+   > **WARNING!** \
+   Druid administrator users have the same OS permissions as the Unix user account running Druid. If the Druid process is running under the OS root user account, then Druid administrators can read or write all files that the root account has access to, including sensitive files such as `/etc/passwd`.
```
Comment (suneet-s) [inline docs/operations/security-overview.md:37]: > Druid administrator

Is there documentation anywhere that talks about who a Druid administrator is. I think users coming from other systems will not be aware that someone who has write privileges to any datasource is considered an administrator. This is a really important fact an operator should be aware of when they are thinking of securing their Druid cluster.

File: docs/operations/security-overview.md [hunk 4]
```diff
@@ -23,66 +23,109 @@ title: "Security overview"
...
+* Enable authentication to the Druid cluster for production environments and other environments that can be accessed by untrusted networks.
+* Do not expose the Druid Console without authentication on untrusted networks. Authenticated Druid Console users have the same permissions as the OS user running the Druid Console process.
```
Comment (suneet-s) [inline docs/operations/security-overview.md:39]: I think this should say authorization instead of authentication. Authenticated users who only have READ DATASOURCE privileges, can use the web console without access to the files on the Druid server.

```suggestion
* Do not expose the Druid Console without authorization enabled. If authorization is not enabled, any user that has access to the web console will have the same permissions as the OS user running the Druid process.
```

File: docs/operations/security-overview.md [hunk 5]
```diff
@@ -23,66 +23,109 @@ title: "Security overview"
...
+* Use an API gateway to restrict access from untrusted networks, create an allow list of specific APIs that your users need to access, and implement account lockout and throttling features.
```
Comment (suneet-s) [inline docs/operations/security-overview.md:40]: What is an account lockout? Are these throttling features something that should be implemented outside of Druid?

File: docs/operations/security-overview.md [hunk 6]
```diff
@@ -23,66 +23,109 @@ title: "Security overview"
...
+* When possible, use firewall and other network layer filtering to only expose Druid services and ports specifically required for your use case. For example, only expose Broker ports to downstream applications that execute queries. You can limit access to a specific IP address or IP range to further tighten and enhance security.
```
Comment (suneet-s) [inline docs/operations/security-overview.md:41]: I think this point should be collapsed with the previous point. They're both talking about restricting access to API end-points via different tools - an API gateway or a firewall

File: docs/operations/security-overview.md [hunk 7]
```diff
@@ -23,66 +23,109 @@ title: "Security overview"
...
+* Only grant `STATE READ`, `STATE WRITE`, and `DATASOURCE WRITE` permissions to highly-trusted users. These permissions allows users to access resources on behalf of the Druid server process regardless of the datasource.
```
Comment (suneet-s) [inline docs/operations/security-overview.md:42]: This should be the point right after "enable authentication". It should be something like "Enable authorization because ..."

Typing out this comment made me realize there's a big overlap between this point and line 39. We'll want to put those 2 points closer to each other I think

File: docs/operations/security-overview.md [hunk 8]
```diff
@@ -23,66 +23,109 @@ title: "Security overview"
...
+* If your Druid client application allows less-trusted users to control the input source or firehose of an ingestion task, validate the URLs from the users. It is possible to point unchecked URLs to other locations and resources within your network or local file system.
+* Enable TLS to encrypt communication within the cluster.
+* You should only grant `WRITE` permissions to any `DATASOURCE` to trusted users. Druid's trust model assumes those users have the same privileges as the operating system user that runs the Druid process.
+* Grant users the minimum permissions necessary to perform their functions. For instance, do not allow users who only need to query data to write to data sources or view state.
```
Comment (suneet-s) [inline docs/operations/security-overview.md:46]: This overlaps with the point about authorization

File: docs/operations/security-overview.md [hunk 9]
```diff
@@ -23,66 +23,109 @@ title: "Security overview"
...
+* Disable JavaScript, as noted in the [Security section](https://druid.apache.org/docs/latest/development/javascript.html#security) of the JavaScript guide.
```
Comment (suneet-s) [inline docs/operations/security-overview.md:47]: This should probably be higher

File: docs/operations/security-overview.md [hunk 10]
```diff
@@ -23,66 +23,109 @@ title: "Security overview"
...
-## Overview
+
+This document provides an overview of Apache Druid security features, configuration instructions, and some best practices to secure Druid.
 
 By default, security features in Druid are disabled, which simplifies the initial deployment experience. However, security features must be configured in a production deployment. These features include TLS, authentication, and authorization.
 
-To implement Druid security, you configure authenticators and authorizers. Authenticators control the way user identities are verified, while authorizers map the authenticated users (via user roles) to the datasources they are permitted to access. Consequently, implementing Druid security also involves considering your datasource scheme, since that scheme represents the granularity at which data access permissions are allocated. 
 
-The following graphic depicts the course of request through the authentication process: 
+## Best practices
```
Comment (suneet-s) [inline docs/operations/security-overview.md:32]: Thanks for putting these best practices together. Reading through them, I made a bunch of comments. Here is some structure that I think will help.

We should break this into a few sections
1. Druid cluster set up
1a. Run as a non root user
1b. Disable javascript
1c. Enable authentication  and authorization
2.  Network setup
2a. Enable TLS
2b. Limit access to the cluster, etc.
3. Authentication + Authorization
3a. Points about authentication and authorization, minimum privileges, etc.

File: docs/operations/security-overview.md [hunk 11]
```diff
@@ -23,66 +23,109 @@ title: "Security overview"
...
+* Grant users the minimum permissions necessary to perform their functions. For instance, do not allow users who only need to query data to write to data sources or view state.
+* Disable JavaScript, as noted in the [Security section](https://druid.apache.org/docs/latest/development/javascript.html#security) of the JavaScript guide.
 
+## Authentication and Authorization
 
-This document gives you an overview of security features in Druid and how to configure them, and some best practices for securing Druid.
+You can configure authentication and authorization to control access to the the Druid APIs. The first step is enabling TLS for the cluster nodes. This is essential to ensure passwords and authentication tokens are encrypted over the network.
+Then configure users, roles, and permissions, as described in the following sections. 
 
+The configuration settings mentioned below are primarily located in the `common.runtime.properties` file. Note that you need to make the configuration changes on all Druid server in the cluster.
```
Comment (suneet-s) [inline docs/operations/security-overview.md:55]: It looks like this title section is accidentally duplicated. More info seems to be written starting on line 121

File: docs/operations/security-overview.md [hunk 12]
```diff
@@ -93,18 +136,28 @@ The following takes you through sample configuration steps for enabling basic au
 
 1. Add the `druid-basic-security` extension to `druid.extensions.loadList` in `common.runtime.properties`. For the quickstart installation, for example, the properties file is at `conf/druid/cluster/_common`:
    ```
-   druid.extensions.loadList=["druid-basic-security", "druid-histogram", "druid-datasketches", "druid-kafka-indexing-service"]
+   druid.extensions.loadList=["druid-basic-security", "druid-histogram", "druid-datasketches", "druid-kafka-indexing-service", "imply-utility-belt"]
```
Comment (suneet-s) [inline docs/operations/security-overview.md:139]:
```suggestion
   druid.extensions.loadList=["druid-basic-security", "druid-histogram", "druid-datasketches", "druid-kafka-indexing-service"]
```

File: docs/operations/security-overview.md [hunk 13]
```diff
@@ -23,66 +23,109 @@ title: "Security overview"
...
+
+# Setup client side TLS
+druid.client.https.protocol=TLSv1.2
+druid.client.https.trustStoreType=jks
+druid.client.https.trustStorePath=truststore.jks # replace with correct turstStore file
+druid.client.https.trustStorePassword=secret123  # replace with your own password
+
+# Setup server side TLS
+druid.server.https.keyStoreType=jks
+druid.server.https.keyStorePath=imply-keystore.jks # replace with correct keyStore file
```
Comment (suneet-s) [inline docs/operations/security-overview.md:113]:
```suggestion
druid.server.https.keyStorePath=my-keystore.jks # replace with correct keyStore file
```

File: docs/operations/security-overview.md [hunk 14]
```diff
@@ -136,37 +189,37 @@ The following steps walk through a sample setup procedure:
...
 4. Assign roles to users by issuing a POST request to `druid-ext/basic-security/authorization/db/MyBasicMetadataAuthorizer/users/<USERNAME>/roles/<ROLENAME>`. For example: 
 	```
-	curl -u admin:password -XPOST https://my-coordinator-ip:8281/druid-ext/basic-security/authorization/db/basic/users/myname/roles/myrole | jq
+	curl -u admin:password1 -XPOST https://my-coordinator-ip:8281/druid-ext/basic-security/authorization/db/basic/users/myname/roles/myrole | jq
 	```
 5. Finally, attach permissions to the roles to control how they can interact with Druid at `druid-ext/basic-security/authorization/db/MyBasicMetadataAuthorizer/roles/<ROLENAME>/permissions`. 
 	For example: 
 	```
-	curl -u admin:password -H'Content-Type: application/json' -XPOST --data-binary @perms.json https://my-coordinator-ip:8281/druid-ext/basic-security/authorization/db/basic/roles/myrole/permissions
+	curl -u admin:password1 -H'Content-Type: application/json' -XPOST --data-binary @perms.json https://my-coordinator-ip:8281/druid-ext/basic-security/authorization/db/basic/roles/myrole/permissions
```
Comment (suneet-s) [inline docs/operations/security-overview.md:222]: why the change to `password1`?

File: docs/operations/security-overview.md [hunk 15]
```diff
@@ -264,3 +318,29 @@ As an alternative to using the basic metadata authenticator, as shown in the pre
 
 
 Congratulations, you have configured permissions for user-assigned roles in Druid!
+
+
+## Druid security trust model
+Like all other security systems, trust is the foundation of the Druid security model. Druid administrators and read-only users are trusted users. Therefore, they are not expected to act maliciously.
```
Comment (suneet-s) [inline docs/operations/security-overview.md:324]: I think it's reasonable to expect administrators to be trusted users, but the requirement that read-only act in a non-malicious way is not necessarily true.

File: docs/operations/security-overview.md [hunk 16]
```diff
@@ -264,3 +318,29 @@ As an alternative to using the basic metadata authenticator, as shown in the pre
...
+
+
+Based on this expectation, Druid operates according to the following principles:
+
+From the inner most layer:
+1. Druid processes run within the system user context. They have access to the local files granted to the specified system user.
+2. The Druid ingestion system can create new processes to execute tasks. Those tasks inherit the user of their parent process. This means that any user authorized to submit an ingestion task can use the ingestion task permissions to read or write any local files that the Druid process has access to.
+
```
Comment (suneet-s) [inline docs/operations/security-overview.md:332]: When a druid cluster is configured to talk to an external resource (for example S3, an internal service, etc.) The user submitting the task also has permissions to talk to the external resource

File: docs/operations/security-overview.md [hunk 17]
```diff
@@ -264,3 +318,29 @@ As an alternative to using the basic metadata authenticator, as shown in the pre
...
+> Note: Only grant the permission to submit ingestion tasks to trusted users because they can read and write to local file system.
+
+Within the cluster:
+1. Druid assumes it operates on an isolated, protected network where no reachable IP within the network is under adversary control. It is the responsibility of system implementers to setup firewalls and other methods of protection.
+2. Druid supports TLS encryption for network traffic, including API calls and data transfers.
```
Comment (suneet-s) [inline docs/operations/security-overview.md:337]: I don't think TLS encryption is part of the trust model.

File: docs/operations/security-overview.md [hunk 18]
```diff
@@ -264,3 +318,29 @@ As an alternative to using the basic metadata authenticator, as shown in the pre
...
+Based on this expectation, Druid operates according to the following principles:
+
+From the inner most layer:
+1. Druid processes run within the system user context. They have access to the local files granted to the specified system user.
+2. The Druid ingestion system can create new processes to execute tasks. Those tasks inherit the user of their parent process. This means that any user authorized to submit an ingestion task can use the ingestion task permissions to read or write any local files that the Druid process has access to.
+
+> Note: Only grant the permission to submit ingestion tasks to trusted users because they can read and write to local file system.
+
+Within the cluster:
+1. Druid assumes it operates on an isolated, protected network where no reachable IP within the network is under adversary control. It is the responsibility of system implementers to setup firewalls and other methods of protection.
```
Comment (suneet-s) [inline docs/operations/security-overview.md:336]: This is a little vague. Is the operator supposed to protect incoming connections to Druid or outgoing or both?

As it's written it seems like the expectation is that restrictions should be set up for both incoming and outgoing connections. Most systems I know of primarily concern themselves with incoming connections, but I'm not a security expert.

File: docs/operations/security-overview.md [hunk 19]
```diff
@@ -264,3 +318,29 @@ As an alternative to using the basic metadata authenticator, as shown in the pre
...
+2. Druid supports TLS encryption for network traffic, including API calls and data transfers.
+3. Druid assumes auxiliary services such as the metadata store and ZooKeeper nodes are not under adversary control.
+
+Cluster to deep storage:
+1. Druid does not make assumptions about the security for deep storage. It follows the system's native security policies to authenticate and authorize with deep storage.
+2. Druid does not encrypt files for deep storage. Instead, it relies on the storage system's native encryption capabilities to ensure compatibility with encryption schemes across all storage types.
+
+Cluster to client:
+1. Druid authenticates with the client based on the configured authenticator.
+2. Druid only executes queries when an authorizer grants permission. The default configuration is `allowAll authorizer`.
```
Comment (suneet-s) [inline docs/operations/security-overview.md:346]: Slight suggestion since it's not just queries, but even things like loading data, reading configuration, changing state, etc. The authorizer must grant permissions for any of these actions to be performed.

```suggestion
2. Druid only performs actions when an authorizer grants permission. The default configuration is `allowAll authorizer`.
```

File: docs/operations/security-overview.md (hunk 5 above)
Comment (2bethere) [reply docs/operations/security-overview.md:40]: Yes, those are features provided by API gateways.

File: docs/operations/security-overview.md (hunk 15 above)
Comment (2bethere) [reply docs/operations/security-overview.md:324]: Actually, I think any authenticated user is expected to not act in a non-malicious way. If you are worried about those users, you should build another layer on top of Druid?

File: docs/operations/security-overview.md (hunk 17 above)
Comment (2bethere) [reply docs/operations/security-overview.md:337]: Any suggestions on how we describe that the network traffic is protected?

File: docs/operations/security-overview.md (hunk 3 above)
Comment (2bethere) [reply docs/operations/security-overview.md:37]:
```suggestion
Users with write privileges to any data source are considered as a Druid administrator. Druid administrator users have the same OS permissions as the Unix user account running Druid. If the Druid process is running under the OS root user account, then Druid administrators can read or write all files that the root account has access to, including sensitive files such as `/etc/passwd`.
```

File: docs/operations/security-overview.md (hunk 5 above)
Comment (techdocsmith) [reply docs/operations/security-overview.md:40]: Everything in that list should be in the API gateway. Might be clearer with sub-bullets.

File: docs/operations/security-overview.md (hunk 17 above)
Comment (techdocsmith) [reply docs/operations/security-overview.md:337]: @2bethere , @suneet-s TLS is "A" trust model if not "Druid's" model. We have already mentioned in Best Practices section to Enable TLS. So maybe we are covered on this front?

File: docs/operations/security-overview.md (hunk 18 above)
Comment (techdocsmith) [reply docs/operations/security-overview.md:336]: I think you may need to secure incoming and outgoing, no? If Druid runs as a user that could have permissions may have access to other network resources. This almost begs the question of do we need a security whitepaper type doc with illustrative examples of the best practices. (Not the goal of this PR).

Here's my attempt to clarify FWIW:
> Druid assumes it operates on an isolated, protected network where no reachable IP within the network is under adversary control. When you implement Druid, take care to setup firewalls and other security measures to secure both inbound and outbound connections.

File: docs/operations/security-overview.md (hunk 15 above)
Comment (techdocsmith) [reply docs/operations/security-overview.md:327]: What can a read-only user do to attack? Asking for my own knowledge. Like if I can't write something, I can't submit malicious code or other, right? (I may be naive on this front).

File: docs/operations/security-overview.md (hunk 17 above)
Comment (2bethere) [reply docs/operations/security-overview.md:340]: Ha, I see. Maybe this:
```suggestion
2. Druid assuming network traffic within the cluster is encrypted, including API calls and data transfers. By default, this is implemented via TLS.
```

Comment (techdocsmith) [reply docs/operations/security-overview.md:340]:
```suggestion
Druid assumes network traffic within the cluster is encrypted, including API calls and data transfers. The default encryption implementation uses TLS.
```

File: docs/operations/security-overview.md [hunk 20]
```diff
@@ -264,3 +321,29 @@ As an alternative to using the basic metadata authenticator, as shown in the pre
...
+Like all other security systems, trust is the foundation of the Druid security model. Druid administrators and read-only users are trusted users. Therefore, they are not expected to act maliciously.
+
+
+Based on this expectation, Druid operates according to the following principles:
+
+From the inner most layer:
+1. Druid processes run within the system user context. They have access to the local files granted to the specified system user.
+2. The Druid ingestion system can create new processes to execute tasks. Those tasks inherit the user of their parent process. This means that any user authorized to submit an ingestion task can use the ingestion task permissions to read or write any local files or external resources that the Druid process has access to.
+
+> Note: Only grant the permission to submit ingestion tasks to trusted users because they can read and write to local file system.
```
Comment (suneet-s) [inline docs/operations/security-overview.md:336]: It's more than just the filesystem, they can make network calls as well.

```suggestion
> Note: Only grant the permission to submit ingestion tasks to trusted users because they can act as the Druid process.
```

File: docs/operations/security-overview.md (hunk 6 above)
Comment (techdocsmith) [reply docs/operations/security-overview.md:44]: Now that we have grouped these, I think it is OK to keep them as separate points.

File: docs/operations/security-overview.md [hunk 21]
```diff
@@ -23,66 +24,112 @@ title: "Security overview"
...
+   > **WARNING!** \
+   Druid administrators have the same OS permissions as the Unix user account running Druid. See [Authentication and authorization model](security-user-auth.md#authentication-and-authorization-model). If the Druid process is running under the OS root user account, then Druid administrators can read or write all files that the root account has access to, including sensitive files such as `/etc/passwd`.
+* Enable authentication to the Druid cluster for production environments and other environments that can be accessed by untrusted networks.
+* Enable authorization and do not expose the Druid Console without authorization enabled. If authorization is not enabled, any user that has access to the web console has the same privileges as the operating system user that runs the Druid Console process.
+* Grant users the minimum permissions necessary to perform their functions. For instance, do not allow users who only need to query data to write to data sources or view state.
+* * Disable JavaScript, as noted in the [Security section](https://druid.apache.org/docs/latest/development/javascript.html#security) of the JavaScript guide.
```
Comment (suneet-s) [inline docs/operations/security-overview.md:42]:
```suggestion
* Disable JavaScript, as noted in the [Security section](https://druid.apache.org/docs/latest/development/javascript.html#security) of the JavaScript guide.
```

File: docs/operations/security-overview.md [hunk 22]
```diff
@@ -23,66 +24,112 @@ title: "Security overview"
...
+
+The following recommendations apply the network where Druid runs:
```
Comment (suneet-s) [inline docs/operations/security-overview.md:44]:
```suggestion
The following recommendations apply to the network where Druid runs:
```

File: docs/operations/security-overview.md [hunk 23]
```diff
@@ -23,66 +24,112 @@ title: "Security overview"
...
+* Enable TLS to encrypt communication within the cluster.
+* Use an API gateway to:
+  - Restrict access from untrusted networks
+  - Create an allow list of specific APIs that your users need to access
+  - Implement account lockout and throttling features.
+* When possible, use firewall and other network layer filtering to only expose Druid services and ports specifically required for your use case. For example, only expose Broker ports to downstream applications that execute queries. You can limit access to a specific IP address or IP range to further tighten and enhance security.
```
Comment (suneet-s) [inline docs/operations/security-overview.md:50]: The previous bullet and this one make the same point, just with different technologies. I think we should collapse them and tell operators they can use either an API gateway, or a firewall or something else...

File: docs/operations/security-overview.md [hunk 24]
```diff
@@ -136,37 +193,37 @@ The following steps walk through a sample setup procedure:
 
 > The default Coordinator API port is 8081 for non-TLS connections and 8281 for secured connections.
 
-1. Create a user by issuing a POST request to `druid-ext/basic-security/authentication/db/MyBasicMetadataAuthenticator/users/<USERNAME>`, replacing USERNAME with the new username. For example: 
+1. Create a user by issuing a POST request to `druid-ext/basic-security/authentication/db/MyBasicMetadataAuthenticator/users/<USERNAME>`, replacing USERNAME with the *new* username you are trying to create. For example:
```
Comment (suneet-s) [inline docs/operations/security-overview.md:196]: This example talks about using the druid basic security extension. The rest of this doc talks about principles that apply across Druid. It feels like we want to separate this into separate pages.

Since this example already existed in this page, I don't think we need to move it in this PR, but perhaps in a follow up one.

File: docs/operations/security-overview.md [hunk 25]
```diff
@@ -264,3 +322,31 @@ As an alternative to using the basic metadata authenticator, as shown in the pre
 
 
 Congratulations, you have configured permissions for user-assigned roles in Druid!
+
+
+## Druid security trust model
+Within Druid's trust model there users can have different authorization levels:
+- Users with resource write permissions can are allowed to anything that the druid process can do.
```
Comment (suneet-s) [inline docs/operations/security-overview.md:329]:
```suggestion
- Users with resource write permissions can are allowed to do anything that the druid process can do.
```

**Review by suneet-s — COMMENTED** [review]
Some small typos, but otherwise LGTM

File: docs/operations/security-overview.md (hunk 25 above)
Comment (jihoonson) [reply docs/operations/security-overview.md:329]: Should `can` also be eliminated? Like, `Users with resource write permissions are allowed to do`

File: docs/operations/security-overview.md [hunk 26]
```diff
@@ -23,66 +24,112 @@ title: "Security overview"
...
+* Enable TLS to encrypt communication within the cluster.
+* Use an API gateway to:
+  - Restrict access from untrusted networks
+  - Create an allow list of specific APIs that your users need to access
+  - Implement account lockout and throttling features.
+* When possible, use firewall and other network layer filtering to only expose Druid services and ports specifically required for your use case. For example, only expose Broker ports to downstream applications that execute queries. You can limit access to a specific IP address or IP range to further tighten and enhance security.
+
+The following recommendation applies to Druids authorization and authentication model:
+* Only grant `WRITE` permissions to any `DATASOURCE` to trusted users. Druid's trust model assumes those users have the same privileges as the operating system user that runs the Druid Console process. 
+* Only grant `STATE READ`, `STATE WRITE`, and `DATASOURCE WRITE` permissions to highly-trusted users. These permissions allows users to access resources on behalf of the Druid server process regardless of the datasource.
```
Comment (jihoonson) [inline docs/operations/security-overview.md:54]: I think `CONFIG WRITE` should be an adminitrator-ish permission too as you can update dynamic system configs such as lookups with it. I'm not sure about `STATE READ` though. What bad things can happen when a malicious user have the `STATE READ` permission?

File: docs/operations/security-overview.md [hunk 27]
```diff
@@ -264,3 +322,31 @@ As an alternative to using the basic metadata authenticator, as shown in the pre
...
+- Authenticated read only users can execute queries against resources to which they have permissions.
+- An authenticated user without any permissions is allowed to execute queries that don't require access to a resource.
+
+Additionally, Druid operates according to the following principles:
+
+From the inner most layer:
+1. Druid processes have the same access to the local files granted to the specified system user running the process.
+2. The Druid ingestion system can create new processes to execute tasks. Those tasks inherit the user of their parent process. This means that any user authorized to submit an ingestion task can use the ingestion task permissions to read or write any local files or external resources that the Druid process has access to.
+
+> Note: Only grant the permission to submit ingestion tasks to trusted users because they can  act as the Druid process.
```
Comment (jihoonson) [inline docs/operations/security-overview.md:339]: Here and [this figure](https://github.com/apache/druid/pull/11016/files#diff-3e8eb443238c8a04b52e7691033cfa2b8bd133611434b548c5cd9eaa3a8a72c3R189) talk about the permission to submit ingestion tasks. This seems ambiguous to me. Maybe it would be better to say the `DATASOURCE WRITE` permission instead.

File: docs/operations/security-overview.md [hunk 28]
```diff
@@ -23,66 +24,112 @@ title: "Security overview"
...
-## Overview
+
+This document provides an overview of Apache Druid security features, configuration instructions, and some best practices to secure Druid.
 
 By default, security features in Druid are disabled, which simplifies the initial deployment experience. However, security features must be configured in a production deployment. These features include TLS, authentication, and authorization.
 
-To implement Druid security, you configure authenticators and authorizers. Authenticators control the way user identities are verified, while authorizers map the authenticated users (via user roles) to the datasources they are permitted to access. Consequently, implementing Druid security also involves considering your datasource scheme, since that scheme represents the granularity at which data access permissions are allocated. 
 
-The following graphic depicts the course of request through the authentication process: 
+## Best practices
```
Comment (jihoonson) [inline docs/operations/security-overview.md:33]: Suggest linking these configs somewhere in this section.

- Ingestion security configs: https://github.com/apache/druid/blob/master/docs/configuration/index.md#ingestion-security-configuration
- JDBC connections security configs: https://github.com/apache/druid/blob/master/docs/configuration/index.md#jdbc-connections-to-external-databases

**Review by jihoonson — COMMENTED** [review]
@techdocsmith this doc is great! I left a couple of comments, but please treat only my comments on the typo and `CONFIG WRITE` as blockers for this PR. Other suggestions can be done as follow-ups.

File: docs/operations/security-overview.md (hunk 23 above)
Comment (techdocsmith) [reply docs/operations/security-overview.md:50]: These are grouped now.

**Review by suneet-s — APPROVED** [review]
